src/zcash/Proof.cpp

```diff
@@ -225,27 +225,65 @@ void initialize_curve_params()
 
 ProofVerifier ProofVerifier::Strict() {
     initialize_curve_params();
-    return ProofVerifier(true);
+    return ProofVerifier(true,false,NULL);
 }
 
+ProofVerifier::ProofVerifier(bool perform_reg_verification,bool perform_batch_verification,void* batch_accumulator) :
+    perform_reg_verification(perform_reg_verification),
+    perform_batch_verification(perform_batch_verification),
+    batch_accumulator(batch_accumulator)
+{ 
+ 
+}
+
+
+ProofVerifier ProofVerifier::Batch() {
+    initialize_curve_params();
+    void* acc = (void *) new batch_verification_accumulator<curve_pp>();
+    return ProofVerifier(false,true,acc);
+}
+ProofVerifier ProofVerifier::RegularAndBatch() {
+    initialize_curve_params();
+    void* acc = (void *) new batch_verification_accumulator<curve_pp>();
+    return ProofVerifier(true,true,acc);
+}
 ProofVerifier ProofVerifier::Disabled() {
     initialize_curve_params();
-    return ProofVerifier(false);
+    return ProofVerifier(false,false,NULL);
 }
 
 template<>
 bool ProofVerifier::check(
     const r1cs_ppzksnark_verification_key<curve_pp>& vk,
     const r1cs_ppzksnark_processed_verification_key<curve_pp>& pvk,
+    const r1cs_ppzksnark_processed_batch_verification_key<curve_pp>& pbvk,
     const r1cs_primary_input<curve_Fr>& primary_input,
     const r1cs_ppzksnark_proof<curve_pp>& proof
 )
 {
-    if (perform_verification) {
-        return r1cs_ppzksnark_online_verifier_strong_IC<curve_pp>(pvk, primary_input, proof);
-    } else {
-        return true;
+    //if it's a batch verifier we don't perform verification but just add another proof to the batch
+    if (perform_batch_verification){
+        auto acc = (batch_verification_accumulator<curve_pp> *)batch_accumulator;
+        r1cs_ppzksnark_batcher<curve_pp>(vk, *acc, primary_input, proof);
     }
+    if (perform_reg_verification) {
+      return r1cs_ppzksnark_online_verifier_strong_IC<curve_pp>(pvk, primary_input, proof);
+    }
+    return true;
```

Comment:
This can actually be left almost unchanged without affecting the logic, and it will make the diff slightly easier to parse:

```diff
-    if (perform_verification) {
+    if (perform_reg_verification) {
         return r1cs_ppzksnark_online_verifier_strong_IC<curve_pp>(pvk, primary_input, proof);
     } else {
         return true;
     }
```